src/PostgREST/Query.hs

```diff
@@ -233,11 +236,19 @@ optionalRollback AppConfig{..} ApiRequest{iPreferences=Preferences{..}} = do
     shouldRollback =
       configDbTxAllowOverride && preferTransaction == Just Rollback
 
+usePipeline :: Bool
+usePipeline = True
+
+queuePipelineStatement :: params -> SQL.Statement params () -> SQL.Transaction ()
+queuePipelineStatement params stmt =
+  if usePipeline then SQL.inTransaction $ Session.queuePipelineStatement params stmt
+                 else SQL.statement params stmt
```

Comment:
So awesome that is such a small change.

> the implementation seems mostly correct, but error handling is at least a bit broken (we don't treat aborted pipelines quite right), and there's a decent chance that some failure scenarios actually mess up the connection state, though I haven't seen that

So since there could be some unknown unknowns(pipeline mode also new to me), I think we should make this configurable. Name could be `db-pipeline-mode`.

Reply:
I mean, the large change is in `hasql`. And I don't think it's a particularly sensible implementation there API-wise so far.

But yes gating this by a configuration option makes sense.